WooCommerce/Classes/Yosemite/DefaultStoresManager.swift

```diff
@@ -294,10 +294,11 @@ private extension DefaultStoresManager {
     func replaceTempCredentialsIfNecessary(account: Account) {
         guard
             let credentials = sessionManager.defaultCredentials,
+            let authToken = credentials.authToken,
             credentials.hasPlaceholderUsername() else {
                 return
         }
-        authenticate(credentials: .init(username: account.username, authToken: credentials.authToken, siteAddress: credentials.siteAddress))
+        authenticate(credentials: .init(username: account.username, authToken: authToken, siteAddress: credentials.siteAddress))
```

Comment:
I tried to dig for the context for which this method is needed and found #804. This method will be needed after authentication completes - currently we are supporting only WPCom login so this makes sense. But with WPOrg supported, we should adjust this method to authenticate with WPOrg too.
```suggestion
        if let authToken = credentials.authToken {
             authenticate(credentials: .init(username: account.username, authToken: authToken, siteAddress: credentials.siteAddress))
        } else if password = credentials.password {
             authenticate(credentials: .init(username: account.username, password: password, siteAddress: credentials.siteAddress))
        }
       
```

Reply:
If I understand the reason correctly, we are authenticating again when `credentials.hasPlaceholderUsername()` is true. And the PR #840 description explains about why a placeholder username is used. 
i.e. The username is fetched using the authentication token and the placeholder username will be replaced. 

But, when it comes to the WPOrg credentials flow, wouldn't we already have the username entered by the user? Will we use a placeholder username in the WPOrg flow as well?

Reply:
From my understanding, we want to set `isWPComLoginRequiredForSiteCredentialsLogin` to false when handling authentication with application password for self-hosted sites - and that [will trigger `sync` delegate method.](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/cb036688d89213b2464ecb8905a965a21e90e002/WordPressAuthenticator/Unified%20Auth/View%20Related/Site%20Address/SiteCredentialsViewController.swift#L488). I can see that we get the username and site address there, so we don't need to check for the WPOrg case here. Could you add a comment regarding that to give a context for why we only check WPCom credential here?

Reply:
> we want to set isWPComLoginRequiredForSiteCredentialsLogin to false when handling authentication with application password for self-hosted sites - and that [will trigger sync delegate method.](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/cb036688d89213b2464ecb8905a965a21e90e002/WordPressAuthenticator/Unified%20Auth/View%20Related/Site%20Address/SiteCredentialsViewController.swift#L488).

I agree. After this we will have to store those credentials (username and password) as `defaultCredentials` from the  `AuthenticationManager` directly. [Instead of this line](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/Authentication/AuthenticationManager.swift#L498)

And yes, I think we should handle this when we work on that task (and have a better picture about this flow)